service/history/timerQueueActiveTaskExecutor.go

```diff
@@ -447,8 +447,8 @@ func (t *timerQueueActiveTaskExecutor) executeActivityRetryTimerTask(
 		}
 		return nil
 	}
-	ok, err = verifyTaskVersion(t.shard, t.logger, namespace.ID(task.NamespaceID), activityInfo.Version, task.Version, task)
-	if err != nil || !ok {
+	ok = verifyTaskVersion(t.shard, t.logger, mutableState.GetNamespaceEntry(), activityInfo.Version, task.Version, task)
+	if !ok {
 		return err
```

Comment:
return err will return nil, if you intend to return nil, then make it clear by return nil